In [6]:
import pandas as pd
from sklearn.datasets import fetch_openml
from sklearn.model_selection import train_test_split

X, y = fetch_openml("mnist_784", version=1, return_X_y=True, as_frame=False)
X = X / 255.0

x_train, x_test, y_train, y_test = train_test_split(X, y, random_state=0, test_size=0.3)

/usr/local/lib/python3.10/dist-packages/sklearn/datasets/_openml.py:968: FutureWarning: The default value of `parser` will change from `'liac-arff'` to `'auto'` in 1.4. You can set `parser='auto'` to silence this warning. Therefore, an `ImportError` will be raised from 1.4 if the dataset is dense and pandas is not installed. Note that the pandas parser may return different data types. See the Notes Section in fetch_openml's API doc for details.
  warn(


In [16]:
from sklearn.neural_network import MLPClassifier

mlp1 = MLPClassifier(hidden_layer_sizes=10, activation="logistic",max_iter=2000)
mlp2 = MLPClassifier(hidden_layer_sizes=100, activation="logistic",max_iter=2000)

In [18]:
mlp1.fit(x_train,y_train)
mlp2.fit(x_train,y_train)

MLPClassifier(activation='logistic', hidden_layer_sizes=100, max_iter=2000)

In [19]:
predicted1 = mlp1.predict(x_test)
predicted2 = mlp2.predict(x_test)


In [21]:
from sklearn.metrics import classification_report

print("**10 hidden layers***")
print(classification_report(y_test,predicted1))

print("**100 hidden layers***")
print(classification_report(y_test,predicted2))

**10 hidden layers***
              precision    recall  f1-score   support

           0       0.97      0.96      0.96      2077
           1       0.95      0.98      0.96      2385
           2       0.93      0.90      0.92      2115
           3       0.89      0.90      0.90      2117
           4       0.91      0.93      0.92      2004
           5       0.91      0.89      0.90      1900
           6       0.94      0.94      0.94      2045
           7       0.93      0.93      0.93      2189
           8       0.89      0.88      0.89      2042
           9       0.91      0.90      0.90      2126

    accuracy                           0.92     21000
   macro avg       0.92      0.92      0.92     21000
weighted avg       0.92      0.92      0.92     21000

**100 hidden layers***
              precision    recall  f1-score   support

           0       0.99      0.98      0.99      2077
           1       0.98      0.99      0.99      2385
           2       0.97      0.98

with 10 hidden layers accuracy -> 0.92
with 100 hidden layers accuracy -> 0.97

in this case, 100 layers gonna be selected